src/app/delegation_backend/src/payload_v1.go

```diff
@@ -0,0 +1,92 @@
+package delegation_backend
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/btcsuite/btcutil/base58"
+	"golang.org/x/crypto/blake2b"
+)
+
+type submitRequestDataV1 struct {
```

Comment:
The reasoning behind versioning is that we want to maintain backward and forward compatibility between the node and the backend. Adding new fields as optional would work in a scenario where we expect to update the backend first and the node later, but not the other way around. For instance, if we ship the node changes but the backend cannot be updated now, the backend cannot verify the signature because it doesn't expect the new field.

Adding version negotiating avoids hiding this under a feature flag or node operators having to update the service URL when the backend is updated and gives control of what version the node should send to the backend operator.

Reply:
As of now, it seems to me that versioning changes are completely unnecessary for adding the commit-hash field. And I'm wary of adding some future-proof negotiation unless we have immediate plan to use and there will be no easier solution at hand

Reply:
@georgeee thanks for comments 

I see reasons why we need both versions e.g. test transition from one to another or have no-downtime solution. Also, if we invest any more effort in this project, we need integration test for proxy and validation tool for persisted data.

How I envisioned that was more of having `v1` along `v0` only when enabled by feature flag. This is the easiest way to handle in the backend. 

Currently, in this PR, we have the double request. The first to get capability and the second to execute post. It actually is not ideal for us because it would require some sort of sticky session in LB. That said, we would like to fully migrate to new version before it's deployed -- as you say, we control the backend -- so it doesn't affect us.

Reply:
@georgeee I made some changes so that we maintain just one version of the payload, without negotiation, having the new field as optional and hiding it in the node behind a feature flag. Please, take another look at the changes.